includes/MediaWiki/Specials/SpecialWikibaseReconcileEdit.php

```diff
@@ -0,0 +1,149 @@
+		parent::__construct( 'WikibaseReconcileEdit' );
+
+		$this->editRequestParser = WikibaseReconcileEditServices::getEditRequestParser();
+		$this->editRequestSaver = WikibaseReconcileEditServices::getEditRequestSaver();
+
+		$propertyId = new PropertyId( 'P1' );
+
+		$this->reconcileDefaultJSON = json_encode( [
+			EditRequestParser::VERSION_KEY => '0.0.1',
+			'urlReconcile' => $propertyId->serialize(),
+		], JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES );
+
+		$this->entityDefaultJSON = json_encode( [
+			EditRequestParser::VERSION_KEY => '0.0.1/minimal',
+			'statements' => [
+				[
+					'property' => $propertyId->serialize(),
+					'value' => "http://example.com/",
+				],
+			],
+		], JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES );
+	}
+
+	public static function factory(): self {
+		return new self();
+	}
+
+	public function getDescription(): string {
+		return $this->msg( 'wikibasereconcileedit-special-description' )->text();
+	}
+
+	protected function getGroupName(): string {
+		return 'wikibase';
+	}
+
+	public function execute( $subPage ) {
+		parent::execute( $subPage );
+		$this->requireLogin();
+		$this->showRequestForm();
+
+		$out = $this->getOutput();
+		$out->addModules( 'ext.WikibaseReconcileEdit' );
+	}
+
+	public function sendRequest( $data ) {
+		$data['reconcile'] = json_decode( $data['reconcile'], true );
+		$data['entity'] = json_decode( $data['entity'], true );
+
+		try {
+			$request = $this->editRequestParser->parseRequestBody( $data );
+			$status = $this->editRequestSaver->persistEdits(
+				[ $request ],
+				$data['token'],
+				$this->getContext()->getUser()
+			);
+		} catch ( LocalizedHttpException $exception ) {
+			return Status::newFatal( \Message::newFromKey( $exception->getMessageValue()->getKey() ) );
+		}
+
+		/** @var EntityRevision $entityRevision */
+		$entityRevision = $status->getValue()[0]['revision'];
+
+		$entityUrl = WikibaseRepo::getDefaultInstance()
+			->getEntityUrlLookup()
+			->getLinkUrl( $entityRevision->getEntity()->getId() );
+
+		if ( $entityUrl ) {
+			$this->getOutput()->redirect( $entityUrl );
+		}
+	}
+
+	private function showRequestForm() {
+		$form = HTMLForm::factory(
+			'ooui',
+			$this->getFormFields(),
+			$this->getContext(),
+		);
+
+		$form->setSubmitCallback( [ $this, 'sendRequest' ] );
+
+		$form->show();
+	}
+
+	private function getFormFields(): array {
+		$fields = [];
+
+		$fields['token'] = [
+			'type' => 'hidden',
+			'maxlength' => 255,
+			'required' => true,
+			'default' => $this->getContext()->getUser()->getEditToken(),
+			'name' => 'token'
+		];
+
+		$fields['reconcile'] = [
+			'type' => 'text',
+			'label-message' => 'wikibasereconcileedit-special-reconcile-label',
+			'maxlength' => 255,
+			'required' => true,
+			'default' => $this->reconcileDefaultJSON,
+			'name' => 'reconcile'
+
+		];
+
+		$fields['entity'] = [
+			'type' => 'textarea',
+			'label-message' => 'wikibasereconcileedit-special-entity-label',
+			'maxlength' => 255,
```

Comment:
Hm, this feels pretty short, to be honest… do we need to limit it this much?

Reply:
done